src/live_pixel_validator.mjs

```diff
@@ -10,29 +10,53 @@ import { ROOT_PREFIX } from './constants.mjs';
  * @typedef {import('./params_validator.mjs').ParamsValidator} ParamsValidator
  */
 
+const EXP_PIXEL_PREFIX_LEN = 3;
+
 export class LivePixelsValidator {
     #compiledPixels;
     #defsVersion;
     #defsVersionKey;
     #forceLowerCase;
 
+    #commonExperimentParamsSchema;
+    #commonExperimentSuffixesSchema;
+    #compiledExperiments;
+
     undocumentedPixels = new Set();
     pixelErrors = {};
 
     /**
      * @param {object} tokenizedPixels similar in format to schemas/pixel_schema.json5.
      * See tests/test_data/valid/expected_processing_results/tokenized_pixels.json for an example.
      * @param {ProductDefinition} productDef
-     * @param {object} ignoreParams contains params that follow the schemas/param_schema.json5 type.
+     * @param {object} experimentsDef experiment definitions, following schemas/native_experiments_schema.json5 type.
      * @param {ParamsValidator} paramsValidator
      */
-    constructor(tokenizedPixels, productDef, ignoreParams, paramsValidator) {
+    constructor(tokenizedPixels, productDef, experimentsDef, paramsValidator) {
         this.#forceLowerCase = productDef.forceLowerCase;
         this.#defsVersion = this.#getNormalizedVal(productDef.target.version);
         this.#defsVersionKey = this.#getNormalizedVal(productDef.target.key);
 
-        this.#compileDefs(tokenizedPixels, ignoreParams, paramsValidator);
+        this.#compileDefs(tokenizedPixels, paramsValidator);
         this.#compiledPixels = tokenizedPixels;
+
+        // Experiments params and suffixes
+        this.#commonExperimentParamsSchema = paramsValidator.compileCommonExperimentParamsSchema();
+        this.#commonExperimentSuffixesSchema = paramsValidator.compileSuffixesSchema(
+            experimentsDef.defaultSuffixes || [],
+            EXP_PIXEL_PREFIX_LEN,
+        );
+
+        // Experiment metrics
+        this.#compiledExperiments = experimentsDef.activeExperiments || {};
+        const defaultsSchema = paramsValidator.compileExperimentMetricSchema({ enum: [1, 2, 3, 4, 5, 6, 10, 11, 21, 30] });
```

Comment:
According to https://app.asana.com/1/137249556945/project/1208889145294658/task/1208747415972724?focus=true, the values used for app_use and search pixels are only `1,4,6,11,21,30` (which is also the case in [the code](https://github.com/duckduckgo/Android/blob/develop/feature-toggles/feature-toggles-impl/src/main/java/com/duckduckgo/feature/toggles/impl/metrics/AppUseMetricPixelsPlugin.kt)), so this definition seems to be looser than that.

Reply:
Ah, didn't realize that. Thanks for catching - fixed!